pip is unreliable, recommend apt/etc instead.

> bitcoinlib could not be located on your system. Please install it using the following command: pip install bitcoinlib --user

```
> sudo apt install python-pip
[sudo] password for user: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libexpat1-dev libpython-all-dev libpython-dev libpython2.7-dev python-all
  python-all-dev python-crypto python-dbus python-dev python-keyring
  python-keyrings.alt python-pip-whl python-secretstorage python-setuptools
  python-wheel python2.7-dev
Suggested packages:
  python-crypto-doc python-dbus-dbg python-dbus-doc gir1.2-gnomekeyring-1.0
  python-fs python-gdata python-keyczar python-secretstorage-doc
  python-setuptools-doc
The following NEW packages will be installed:
  libexpat1-dev libpython-all-dev libpython-dev libpython2.7-dev python-all
  python-all-dev python-crypto python-dbus python-dev python-keyring
  python-keyrings.alt python-pip python-pip-whl python-secretstorage
  python-setuptools python-wheel python2.7-dev
0 upgraded, 17 newly installed, 0 to remove and 11 not upgraded.
Need to get 31.2 MB of archives.
After this operation, 49.3 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 libexpat1-dev amd64 2.2.5-3 [122 kB]
Get:2 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 libpython2.7-dev amd64 2.7.15~rc1-1 [28.2 MB]
Get:3 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 libpython-dev amd64 2.7.15~rc1-1 [7,684 B]
Get:4 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 libpython-all-dev amd64 2.7.15~rc1-1 [1,092 B]
Get:5 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-all amd64 2.7.15~rc1-1 [1,076 B]
Get:6 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python2.7-dev amd64 2.7.15~rc1-1 [286 kB]
Get:7 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-dev amd64 2.7.15~rc1-1 [1,256 B]
Get:8 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-all-dev amd64 2.7.15~rc1-1 [1,100 B]
Get:9 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-crypto amd64 2.6.1-8ubuntu2 [244 kB]
Get:10 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-dbus amd64 1.2.6-1 [90.2 kB]
Get:11 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-secretstorage all 2.3.1-2 [11.8 kB]
Get:12 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-keyring all 10.6.0-1 [30.6 kB]
Get:13 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-keyrings.alt all 3.0-1 [16.7 kB]
Get:14 http://ca.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python-pip-whl all 9.0.1-2.3~ubuntu1 [1,652 kB]
Get:15 http://ca.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python-pip all 9.0.1-2.3~ubuntu1 [151 kB]
Get:16 http://ca.archive.ubuntu.com/ubuntu bionic/main amd64 python-setuptools all 39.0.1-2 [329 kB]
Get:17 http://ca.archive.ubuntu.com/ubuntu bionic/universe amd64 python-wheel all 0.30.0-0.2 [36.4 kB]
Fetched 31.2 MB in 17s (1,789 kB/s)                                            
Selecting previously unselected package libexpat1-dev:amd64.
(Reading database ... 179557 files and directories currently installed.)
Preparing to unpack .../00-libexpat1-dev_2.2.5-3_amd64.deb ...
Unpacking libexpat1-dev:amd64 (2.2.5-3) ...
Selecting previously unselected package libpython2.7-dev:amd64.
Preparing to unpack .../01-libpython2.7-dev_2.7.15~rc1-1_amd64.deb ...
Unpacking libpython2.7-dev:amd64 (2.7.15~rc1-1) ...
Selecting previously unselected package libpython-dev:amd64.
Preparing to unpack .../02-libpython-dev_2.7.15~rc1-1_amd64.deb ...
Unpacking libpython-dev:amd64 (2.7.15~rc1-1) ...
Selecting previously unselected package libpython-all-dev:amd64.
Preparing to unpack .../03-libpython-all-dev_2.7.15~rc1-1_amd64.deb ...
Unpacking libpython-all-dev:amd64 (2.7.15~rc1-1) ...
Selecting previously unselected package python-all.
Preparing to unpack .../04-python-all_2.7.15~rc1-1_amd64.deb ...
Unpacking python-all (2.7.15~rc1-1) ...
Selecting previously unselected package python2.7-dev.
Preparing to unpack .../05-python2.7-dev_2.7.15~rc1-1_amd64.deb ...
Unpacking python2.7-dev (2.7.15~rc1-1) ...
Selecting previously unselected package python-dev.
Preparing to unpack .../06-python-dev_2.7.15~rc1-1_amd64.deb ...
Unpacking python-dev (2.7.15~rc1-1) ...
Selecting previously unselected package python-all-dev.
Preparing to unpack .../07-python-all-dev_2.7.15~rc1-1_amd64.deb ...
Unpacking python-all-dev (2.7.15~rc1-1) ...
Selecting previously unselected package python-crypto.
Preparing to unpack .../08-python-crypto_2.6.1-8ubuntu2_amd64.deb ...
Unpacking python-crypto (2.6.1-8ubuntu2) ...
Selecting previously unselected package python-dbus.
Preparing to unpack .../09-python-dbus_1.2.6-1_amd64.deb ...
Unpacking python-dbus (1.2.6-1) ...
Selecting previously unselected package python-secretstorage.
Preparing to unpack .../10-python-secretstorage_2.3.1-2_all.deb ...
Unpacking python-secretstorage (2.3.1-2) ...
Selecting previously unselected package python-keyring.
Preparing to unpack .../11-python-keyring_10.6.0-1_all.deb ...
Unpacking python-keyring (10.6.0-1) ...
Selecting previously unselected package python-keyrings.alt.
Preparing to unpack .../12-python-keyrings.alt_3.0-1_all.deb ...
Unpacking python-keyrings.alt (3.0-1) ...
Selecting previously unselected package python-pip-whl.
Preparing to unpack .../13-python-pip-whl_9.0.1-2.3~ubuntu1_all.deb ...
Unpacking python-pip-whl (9.0.1-2.3~ubuntu1) ...
Selecting previously unselected package python-pip.
Preparing to unpack .../14-python-pip_9.0.1-2.3~ubuntu1_all.deb ...
Unpacking python-pip (9.0.1-2.3~ubuntu1) ...
Selecting previously unselected package python-setuptools.
Preparing to unpack .../15-python-setuptools_39.0.1-2_all.deb ...
Unpacking python-setuptools (39.0.1-2) ...
Selecting previously unselected package python-wheel.
Preparing to unpack .../16-python-wheel_0.30.0-0.2_all.deb ...
Unpacking python-wheel (0.30.0-0.2) ...
Setting up python-pip-whl (9.0.1-2.3~ubuntu1) ...
Setting up python-setuptools (39.0.1-2) ...
Setting up python-crypto (2.6.1-8ubuntu2) ...
Setting up python-wheel (0.30.0-0.2) ...
Setting up python-keyrings.alt (3.0-1) ...
Setting up libexpat1-dev:amd64 (2.2.5-3) ...
Processing triggers for man-db (2.8.3-2) ...
Setting up libpython2.7-dev:amd64 (2.7.15~rc1-1) ...
Setting up python-dbus (1.2.6-1) ...
Setting up python-pip (9.0.1-2.3~ubuntu1) ...
Setting up python2.7-dev (2.7.15~rc1-1) ...
Setting up python-all (2.7.15~rc1-1) ...
Setting up python-secretstorage (2.3.1-2) ...
Setting up libpython-dev:amd64 (2.7.15~rc1-1) ...
Setting up python-keyring (10.6.0-1) ...
Setting up python-dev (2.7.15~rc1-1) ...
Setting up libpython-all-dev:amd64 (2.7.15~rc1-1) ...
Setting up python-all-dev (2.7.15~rc1-1) ...

> pip install bitcoinlib --user
Collecting bitcoinlib
  Downloading https://files.pythonhosted.org/packages/26/51/8e99b9fff9cf63e9caeb37aa84be677ce13c2609d8f686d963c2e52efd30/bitcoinlib-0.3.37.tar.gz (318kB)
    100% |████████████████████████████████| 327kB 1.9MB/s 
Collecting SQLAlchemy==1.2.5 (from bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/7e/57/435347429f4ff5016b72c9b179c70a6cc67e5f9b4d3f3e9b51362c40b356/SQLAlchemy-1.2.5.tar.gz (5.6MB)
    100% |████████████████████████████████| 5.6MB 227kB/s 
Collecting ecdsa==0.13 (from bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/63/f4/73669d51825516ce8c43b816c0a6b64cd6eb71d08b99820c00792cb42222/ecdsa-0.13-py2.py3-none-any.whl (86kB)
    100% |████████████████████████████████| 92kB 3.1MB/s 
Collecting enum34 (from bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting pyaes==1.6.1 (from bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/44/66/2c17bae31c906613795711fc78045c285048168919ace2220daa372c7d72/pyaes-1.6.1.tar.gz
Collecting requests==2.18.4 (from bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/49/df/50aa1999ab9bde74656c2919d9c0c085fd2b3775fd3eca826012bef76d8c/requests-2.18.4-py2.py3-none-any.whl (88kB)
    100% |████████████████████████████████| 92kB 3.3MB/s 
Collecting scrypt==0.8.6 (from bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/01/6f/3c8dd0f18f73ceddfbdd606c0c895ebb66748606682d77da3743c7c0c56f/scrypt-0.8.6.tar.gz (52kB)
    100% |████████████████████████████████| 61kB 4.2MB/s 
Collecting idna<2.7,>=2.5 (from requests==2.18.4->bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/27/cc/6dd9a3869f15c2edfab863b992838277279ce92663d334df9ecf5106f5c6/idna-2.6-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 3.6MB/s 
Collecting urllib3<1.23,>=1.21.1 (from requests==2.18.4->bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/63/cb/6965947c13a94236f6d4b8223e21beb4d576dc72e8130bd7880f600839b8/urllib3-1.22-py2.py3-none-any.whl (132kB)
    100% |████████████████████████████████| 133kB 2.4MB/s 
Collecting certifi>=2017.4.17 (from requests==2.18.4->bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/df/f7/04fee6ac349e915b82171f8e23cee63644d83663b34c539f7a09aed18f9e/certifi-2018.8.24-py2.py3-none-any.whl (147kB)
    100% |████████████████████████████████| 153kB 2.6MB/s 
Collecting chardet<3.1.0,>=3.0.2 (from requests==2.18.4->bitcoinlib)
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
    100% |████████████████████████████████| 143kB 2.8MB/s 
Building wheels for collected packages: bitcoinlib, SQLAlchemy, pyaes, scrypt
  Running setup.py bdist_wheel for bitcoinlib ... done
  Stored in directory: /home/user/.cache/pip/wheels/e6/f9/28/880c267b07fb5fad2fd15bff13c3b4bfc95519a5012d069f36
  Running setup.py bdist_wheel for SQLAlchemy ... done
  Stored in directory: /home/user/.cache/pip/wheels/7f/47/19/9d21adcd1f537063d81e00b933df6c58c9cd4d278bfb500bb5
  Running setup.py bdist_wheel for pyaes ... done
  Stored in directory: /home/user/.cache/pip/wheels/bd/cf/7b/ced9e8f28c50ed666728e8ab178ffedeb9d06f6a10f85d6432
  Running setup.py bdist_wheel for scrypt ... error
  Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-8yzg5_/scrypt/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpXNSSVzpip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-2.7
  creating build/lib.linux-x86_64-2.7/scrypt
  copying scrypt/scrypt.py -> build/lib.linux-x86_64-2.7/scrypt
  copying scrypt/__init__.py -> build/lib.linux-x86_64-2.7/scrypt
  running build_ext
  building '_scrypt' extension
  creating build/temp.linux-x86_64-2.7
  creating build/temp.linux-x86_64-2.7/src
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/crypto
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/scryptenc
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/util
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/libcperciva
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/libcperciva/alg
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/libcperciva/crypto
  creating build/temp.linux-x86_64-2.7/scrypt-1.2.1/libcperciva/util
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c src/scrypt.c -o build/temp.linux-x86_64-2.7/src/scrypt.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/lib/crypto/crypto_scrypt_smix_sse2.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/crypto/crypto_scrypt_smix_sse2.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/lib/crypto/crypto_scrypt_smix.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/crypto/crypto_scrypt_smix.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/lib/crypto/crypto_scrypt.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/crypto/crypto_scrypt.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/lib/scryptenc/scryptenc.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/scryptenc/scryptenc.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/lib/scryptenc/scryptenc_cpuperf.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/scryptenc/scryptenc_cpuperf.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/lib/util/memlimit.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/lib/util/memlimit.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/libcperciva/alg/sha256.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/libcperciva/alg/sha256.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/libcperciva/crypto/crypto_aes_aesni.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/libcperciva/crypto/crypto_aes_aesni.o -O2
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-nbjU53/python2.7-2.7.15~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DHAVE_CONFIG_H -DHAVE_CLOCK_GETTIME=1 -DHAVE_LIBRT=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_STRUCT_SYSINFO=1 -DHAVE_STRUCT_SYSINFO_MEM_UNIT=1 -DHAVE_STRUCT_SYSINFO_TOTALRAM=1 -DHAVE_SYSINFO=1 -DHAVE_SYS_SYSINFO_H=1 -D_FILE_OFFSET_BITS=64 -Iscrypt-1.2.1 -Iscrypt-1.2.1/lib -Iscrypt-1.2.1/lib/scryptenc -Iscrypt-1.2.1/lib/crypto -Iscrypt-1.2.1/lib/util -Iscrypt-1.2.1/libcperciva/cpusupport -Iscrypt-1.2.1/libcperciva/alg -Iscrypt-1.2.1/libcperciva/util -Iscrypt-1.2.1/libcperciva/crypto -I/usr/local/include -I/usr/include -I/usr/include/python2.7 -c scrypt-1.2.1/libcperciva/crypto/crypto_aes.c -o build/temp.linux-x86_64-2.7/scrypt-1.2.1/libcperciva/crypto/crypto_aes.o -O2
  scrypt-1.2.1/libcperciva/crypto/crypto_aes.c:6:10: fatal error: openssl/aes.h: No such file or directory
   #include <openssl/aes.h>
            ^~~~~~~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for scrypt
  Running setup.py clean for scrypt
Successfully built bitcoinlib SQLAlchemy pyaes
Failed to build scrypt
Installing collected packages: SQLAlchemy, ecdsa, enum34, pyaes, idna, urllib3, certifi, chardet, requests, scrypt, bitcoinlib
Exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 360, in run
    prefix=options.prefix_path,
  File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/usr/lib/python2.7/dist-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/lib/python2.7/dist-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/usr/lib/python2.7/dist-packages/pip/wheel.py", line 247, in move_wheel_files
    prefix=prefix,
  File "/usr/lib/python2.7/dist-packages/pip/locations.py", line 153, in distutils_scheme
    i.finalize_options()
  File "/usr/share/python-wheels/setuptools-39.0.1-py2.py3-none-any.whl/setuptools/command/install.py", line 38, in finalize_options
    orig.install.finalize_options(self)
  File "/usr/lib/python2.7/distutils/command/install.py", line 357, in finalize_options
    self.create_home_path()
  File "/usr/lib/python2.7/distutils/command/install.py", line 593, in create_home_path
    os.makedirs(path, 0700)
  File "/usr/lib/python2.7/os.py", line 150, in makedirs
    makedirs(head, mode)
  File "/usr/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/home/user/.local/lib/python2.7'

> sudo apt purge python-pip
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libpython-all-dev python-all python-all-dev python-crypto python-dbus python-keyring python-keyrings.alt
  python-pip-whl python-secretstorage python-wheel
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  python-pip*
0 upgraded, 0 newly installed, 1 to remove and 11 not upgraded.
After this operation, 671 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 180263 files and directories currently installed.)
Removing python-pip (9.0.1-2.3~ubuntu1) ...
Processing triggers for man-db (2.8.3-2) ...

> apt search bitcoinlib
Sorting... Done
Full Text Search... Done
python3-bitcoinlib/bionic,bionic 0.9.0-2 all
  Python module with an easy interface to the Bitcoin protocol

> sudo apt install python3-bitcoinlib
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libpython-all-dev python-all python-all-dev python-crypto python-dbus python-keyring python-keyrings.alt
  python-pip-whl python-secretstorage python-wheel
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  libssl-dev libssl-doc
The following NEW packages will be installed:
  libssl-dev libssl-doc python3-bitcoinlib
0 upgraded, 3 newly installed, 0 to remove and 11 not upgraded.
Need to get 2,697 kB of archives.
After this operation, 11.9 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ca.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libssl-dev amd64 1.1.0g-2ubuntu4.1 [1,372 kB]
Get:2 http://ca.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libssl-doc all 1.1.0g-2ubuntu4.1 [1,255 kB]
Get:3 http://ca.archive.ubuntu.com/ubuntu bionic/universe amd64 python3-bitcoinlib all 0.9.0-2 [69.4 kB]
Fetched 2,697 kB in 2s (1,338 kB/s)          
Selecting previously unselected package libssl-dev:amd64.
(Reading database ... 180158 files and directories currently installed.)
Preparing to unpack .../libssl-dev_1.1.0g-2ubuntu4.1_amd64.deb ...
Unpacking libssl-dev:amd64 (1.1.0g-2ubuntu4.1) ...
Selecting previously unselected package libssl-doc.
Preparing to unpack .../libssl-doc_1.1.0g-2ubuntu4.1_all.deb ...
Unpacking libssl-doc (1.1.0g-2ubuntu4.1) ...
Selecting previously unselected package python3-bitcoinlib.
Preparing to unpack .../python3-bitcoinlib_0.9.0-2_all.deb ...
Unpacking python3-bitcoinlib (0.9.0-2) ...
Setting up libssl-dev:amd64 (1.1.0g-2ubuntu4.1) ...
Processing triggers for man-db (2.8.3-2) ...
Setting up libssl-doc (1.1.0g-2ubuntu4.1) ...
Setting up python3-bitcoinlib (0.9.0-2) ...
```

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -1,0 +1,158 @@
+name: tribler-bittorrent
+version: git
+summary: Tribler
+description: >
+  Open source decentralized BitTorrent client which allows anonymous peer-to-peer downloads by default.
+
+grade: devel
+confinement: classic
+base: core18
+
+parts:
+  tribler-deps:
+    plugin: nil
+    stage-packages:
+      - ffmpeg
+      - libssl-dev
+      - libx11-6
+      - vlc
+      - python-apsw
+      - python-cherrypy3
+      - python-keyring
+      - python-keyrings.alt
+      - python-libtorrent
+      - python-m2crypto
+      - python-meliae
+      - python-pil
+      - python-pyqt5
+      - python-pyqt5.qtsvg
+      - python-scipy
+      - python-subprocess32
+      - libxcb-xinerama0
+      - libfontconfig1-dev
+      - libfreetype6-dev
+      - libx11-dev
+      - libxext-dev
+      - libxfixes-dev
+      - libxi-dev
+      - libxrender-dev
+      - libxcb1-dev
+      - libx11-xcb-dev
+      - libxcb-glx0-dev
+      - libxcb-keysyms1-dev
+      - libxcb-image0-dev
+      - libxcb-shm0-dev
+      - libxcb-icccm4-dev
+      - libxcb-sync0-dev
+      - libxcb-xfixes0-dev
+      - libxcb-shape0-dev
+      - libxcb-randr0-dev
+      - libxcb-render-util0-dev
+
+  ipv8-deps:
+    plugin: python
+    python-version: python2
+    source: https://github.com/Tribler/py-ipv8.git
+    source-type: git
+    source-branch: master
+    source-depth: 1
+    requirements: requirements.txt
+    python-packages:
+      - yappi
+      - bitcoinlib
+      - chardet
+      - configobj
+      - decorator
+      - dnspython
+      - ecdsa
+      - feedparser
+      - jsonrpclib
+      - leveldb
+      - matplotlib
+      - netifaces
+      - networkx
+      - pbkdf2
+      - pony
+      - protobuf
+      - psutil
+      - pyaes
+      - pyasn1
+      - pysocks
+      - requests
+      - service_identity
+      - lz4
+    build-packages:
+      - python-dev
+      - python-pip
+      - python-pkg-resources
+      - python-setuptools
+    stage-packages:
+      - python
+      - python-distutils-extra
+
+  tribler:
+    after:
+      - desktop-qt5
+      - tribler-deps
+      - ipv8-deps
+    plugin: python
+    python-version: python2
+    source: https://github.com/Tribler/tribler.git
+    source-type: git
+    source-branch: devel
+    source-depth: 1
+    override-build: |
+      cp -r ./ ${SNAPCRAFT_PART_INSTALL}/usr/share/tribler
+      set -ex
+      for dir in $SNAPCRAFT_PART_INSTALL/usr/lib/*/; do
+        (cd $dir;
+         if [ -f blas/libblas.so.3.* ]; then
+           ln -s blas/libblas.so.3.* libblas.so.3
+         fi
+         if [ -f lapack/liblapack.so.3.* ]; then
+           ln -s lapack/liblapack.so.3.* liblapack.so.3
+         fi
+         )
+      done
+    build-packages:
+      - python-tk
+      - python-faulthandler
+      - python-libtorrent
+      - libblas3
+      - libblas-dev
+      - libxcomposite1
+      - libxcursor1
+      - libxinerama1
+      - libatk1.0-0
+      - libgtk2.0-0
+      - liblapack3
+    stage-packages:
+      - python-tk
+      - python-faulthandler
+      - python-libtorrent
+      - libblas3
+      - libblas-dev
+      - libxcomposite1
+      - libxcursor1
+      - libxinerama1
+      - libatk1.0-0
+      - libgtk2.0-0
+      - liblapack3
+      - libx11-xcb1
+      - libxcb-xkb1
+      - libxcb-icccm4
+      - libxcb-image0
+      - libxcb-render-util0
+
+apps:
+  tribler-bittorrent:
+    command: desktop-launch ${SNAP}/usr/bin/python ${SNAP}/usr/share/tribler/run_tribler.py
+    plugs:
+      - desktop
+      - desktop-legacy
+      - home
+      - network
+      - network-bind
+      - unity7
+      - pulseaudio
+      - network-manager

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -8,6 +8,9 @@ grade: devel
 confinement: classic
 base: core18
 
+architectures:
+  - build-on: amd64
+
 parts:
   tribler-deps:
     plugin: nil

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -12,6 +12,30 @@ architectures:
   - build-on: amd64
 
 parts:
+  desktop-qt5:
+    source: https://github.com/ubuntu/snapcraft-desktop-helpers.git
+    source-subdir: qt
+    plugin: make
+    make-parameters: ["FLAVOR=qt5"]
+    build-packages:
+      - build-essential
+      - qtbase5-dev
+      - dpkg-dev
+    stage-packages:
+      - libxkbcommon0
+      - ttf-ubuntu-font-family
+      - dmz-cursor-theme
+      - light-themes
+      - adwaita-icon-theme
+      - gnome-themes-standard
+      - shared-mime-info
+      - libqt5gui5
+      - libgdk-pixbuf2.0-0
+      - libqt5svg5 # for loading icon themes which are svg
+      - try: [appmenu-qt5] # not available on core18
+      - locales-all
+      - xdg-user-dirs
+      - fcitx-frontend-qt5
   tribler-deps:
     plugin: nil
     stage-packages:

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -1,12 +1,12 @@
 name: tribler-bittorrent
-version: git
-summary: Tribler
+version: latest
+version-script: cat $SNAPCRAFT_STAGE/version
+summary: Decentralized BitTorrent client
 description: >
   Open source decentralized BitTorrent client which allows anonymous peer-to-peer downloads by default.
 
 grade: devel
 confinement: classic
-base: core18
 
 architectures:
   - build-on: amd64
@@ -105,6 +105,9 @@ parts:
     source-branch: devel
     source-depth: 1
     override-build: |
+      VERSION=$(git describe --tags|cut -d- -f1,2)
+      echo $VERSION > $SNAPCRAFT_STAGE/version
+
       cp -r ./ ${SNAPCRAFT_PART_INSTALL}/usr/share/tribler
       set -ex
       for dir in $SNAPCRAFT_PART_INSTALL/usr/lib/*/; do

